Enhance ClaimsPrincipal/Identity content

guardrex opened the pull request:
Fixes #1190 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie?branch=pr-en-us-5669)

Mostly addressed by the significant updates on https://github.com/aspnet/Docs/pull/5307.

* I add a role claim to the sample code.
* I add two related topics.

As for ...

> What claims should I add? Why?

Shouldn't that be left to [Claims-Based Authorization](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/claims)? I link it now in this topic.

anurse commented:
Not sure I have the context on why this change is needed. It looks fine to me, but I think someone like @Tratcher should take a look.

guardrex commented:
@anurse In your [original issue \#1190](https://github.com/aspnet/Docs/issues/1190#issue-153299751), you asked about showing how to set a role claim, so I added that in along with the two name claims. Since the ClaimsPrincipal/Identity was added on the earlier PR (I think that covers your other ask) and a reader might want to know more, I added the two links down to the additional resources.

anurse commented:
Thanks for the context @guardrex ! LGTM
